homeassistant/components/rest/binary_sensor.py

```diff
@@ -98,29 +118,33 @@ def __init__(
         coordinator: DataUpdateCoordinator[None] | None,
         rest: RestData,
         config: ConfigType,
-        unique_id: str | None,
+        trigger_entity_config: ConfigType,
     ) -> None:
         """Initialize a REST binary sensor."""
+        ManualTriggerEntity.__init__(self, hass, trigger_entity_config)
         RestEntity.__init__(
             self,
             coordinator,
             rest,
             config.get(CONF_RESOURCE_TEMPLATE),
             config[CONF_FORCE_UPDATE],
         )
-        TemplateEntity.__init__(
-            self,
-            hass,
-            config=config,
-            fallback_name=DEFAULT_BINARY_SENSOR_NAME,
-            unique_id=unique_id,
-        )
         self._previous_data = None
         self._value_template: Template | None = config.get(CONF_VALUE_TEMPLATE)
         if (value_template := self._value_template) is not None:
             value_template.hass = hass
 
-        self._attr_device_class = config.get(CONF_DEVICE_CLASS)
+    async def async_added_to_hass(self) -> None:
+        """Ensure the data from the initial update is reflected in the state."""
+        await RestEntity.async_added_to_hass(self)
+        await ManualTriggerEntity.async_added_to_hass(self)
```

Comment:
Is this really needed? Is the real bug here that `TriggerBaseEntity.async_added_to_hass` is missing a super call?

Reply:
In previous PR we said that this is a much cleaner way to say what's going on than using super()